### PR TITLE
Add Hex4B rev.2 PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add Hex 4B rev.2 PCB, APM32F103.
Thank you.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/hex4b/rev1/readme.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+![hex4b](https://i.imgur.com/lArfGgHh.jpeg)
+
+A PCB for HEX.4B 75% keyboard.
+
+* Keyboard Maintainer: [Mechlovin' Studio](https://github.com/mechlovin)
+* Hardware Supported: Hex.4B PCB, Atmega32A 
+* Hardware Availability: [Hex Keyboard](https://hexkeyboards.com/)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mechlovin/hex4b/rev1/readme.md:11]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/hex4b/rev1/readme.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
```
Comment (drashna) [inline keyboards/mechlovin/hex4b/rev1/readme.md:16]: whoops, sorry, my fault. 
```suggestion
```

File: keyboards/mechlovin/hex4b/rev1/readme.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    make mechlovin/hex4b/rev1:default
+
+Flashing example for this keyboard:
+
+    make mechlovin/hex4b/rev1:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+Reset Key: To put the Hex4B into bootloader, hold esc key while plugging in (bootloader), hold backspace while plugging in (bootmagic) and plug cable.
```
Comment (drashna) [inline keyboards/mechlovin/hex4b/rev1/readme.md:38]: This should be easier to see, ideally.

File: keyboards/mechlovin/hex4b/rev2/readme.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
```
Comment (drashna) [inline keyboards/mechlovin/hex4b/rev2/readme.md:16]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/hex4b/rev2/rules.mk [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = STM32F103
+
+# Bootloader selection
+BOOTLOADER = stm32duino
+
+
```
Comment (fauxpark) [inline keyboards/mechlovin/hex4b/rev2/rules.mk:7]:
```suggestion
```

File: keyboards/mechlovin/hex4b/readme.md [hunk 6]
```diff
@@ -1,3 +1,20 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
```
Comment (fauxpark) [inline keyboards/mechlovin/hex4b/readme.md:17]:
```suggestion
```

File: keyboards/mechlovin/hex4b/rev1/readme.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (fauxpark) [inline keyboards/mechlovin/hex4b/rev1/readme.md:1]:
```suggestion
```

File: keyboards/mechlovin/hex4b/rev2/readme.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (fauxpark) [inline keyboards/mechlovin/hex4b/rev2/readme.md:1]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**